### PR TITLE
Extended dataset cheatsheet in documentation

### PR DESCRIPTION
**Pull request by EdisonLeeeee**
This should be done in a couple of weeks.

### CODE REVIEW

File: torch_geometric/datasets/md17.py [hunk 1]
```diff
@@ -97,6 +97,115 @@ class MD17(InMemoryDataset):
...
+              - 0
+              - 0
+              - 0
+            * - Benzene
+              - 627,983
+              - 12
+              - 0
+              - 0
+              - 0              
+            * - Benzene CCSD-T
```
Comment (EdisonLeeeee) [inline torch_geometric/datasets/md17.py:124]: Using `CCSD(T)` would result in a building error:
```bash
Handler <function setup.<locals>.rst_jinja_render at 0x1273eb160> for event 'source-read' threw an exception (exception: 'NoneType' object has no attribute 'group')
```
Had to use `CCSD-T` instead.

**Comment by rusty1s** [comment]
The complete one? Wow, that would be amazing!

**Comment by EdisonLeeeee** [comment]
Yep. I saw the [cheatsheet](https://pytorch-geometric.readthedocs.io/en/latest/notes/data_cheatsheet.html) is still a work in progress so I intend to complete them. Given the huge amount of datasets, it might cost me a few weeks to summarize them.

**Comment by rusty1s** [comment]
Not necessary to do them all at once. Just take a few and let‘s split the work :)

**Comment by EdisonLeeeee** [comment]
Jobs done (for this PR) :)

**Review by rusty1s — APPROVED** [review]

**Comment by rusty1s** [comment]
Thank you! There seems to exist a build error. Do you know what‘s the cause of this?

**Comment by EdisonLeeeee** [comment]
Not really sure about this. It's successfully built on my end. Everything seems ok. Weird :(

Any error logs?

**Comment by rusty1s** [comment]
Seems to be a bug on `sphinx==5.2.0`, but `sphinx==5.2.1` fixed it.

**Pull request merged**